### ISSUE TITLE
Embedded functions do not load model weights

### ISSUE THREAD

**Issue by AlKukharskii** [Omid-Nejati/MedViT#10]
Dear developer,

First of all, thank you for your amazing job. It is a wonderful net.
While I was using MedViT, I noticed that it does not load weights. There is the code from medvit.py:

@register_model
def MedViT_small(pretrained=False, pretrained_cfg=None, **kwargs):
    model = MedViT(stem_chs=[64, 32, 64], depths=[3, 4, 10, 3], path_dropout=0.1, **kwargs)
    return model

@register_model
def MedViT_base(pretrained=False, pretrained_cfg=None, **kwargs):
    model = MedViT(stem_chs=[64, 32, 64], depths=[3, 4, 20, 3], path_dropout=0.2, **kwargs)
    return model

@register_model
def MedViT_large(pretrained=False, pretrained_cfg=None, **kwargs):
    model = MedViT(stem_chs=[64, 32, 64], depths=[3, 4, 30, 3], path_dropout=0.2, **kwargs)
    return model

I presume that  it should be something like that:

@register_model
def MedViT_large(pretrained=False, pretrained_cfg=None, **kwargs):
    model = MedViT(stem_chs=[64, 32, 64], depths=[3, 4, 30, 3], path_dropout=0.2, **kwargs)
   if pretrained is True:
       model.load_state_dict(torch.load(pretrained_cfg, weights_only=True)['model'])
    return model

One more time, thank you for your great job!

**Comment by Omid-Nejati** [comment]
Hi, 
1- To use pre-train weights, follow the stracture of this repo : https://github.com/Omid-Nejati/MedViT/blob/main/CustomDataset.md
2- Download pre-train from the table based on model size.
3- Address the location of pre-train weights in this code for finetune : 
cd CustomDataset/
"bash train.sh 8 --model MedViT_small --batch-size 30 --lr 5e-6 --warmup-epochs 0 --weight-decay 1e-8 --epochs 30 --sched step --decay-epochs 60 --input-size 384 --resume ../checkpoints/MedViT_small_im1k.pth --finetune --data-path your_data_path "

**Comment by AlKukharskii** [comment]
Hi!

Please, take a look on the code, it does not load pretrained weights:
Line 515 of the file MedViT.py:
**********************CODE***********************************************
def MedViT_large(pretrained=False, pretrained_cfg=None, **kwargs):
    model = MedViT(stem_chs=[64, 32, 64], depths=[3, 4, 30, 3], path_dropout=0.2, **kwargs)
    return model
**********************END OF CODE****************************************
it takes a file with the pretrained weights, but it does nothing with it. The same is true for MedViT_large and MedViT_base.

**Comment by Omid-Nejati** [comment]
Thanks for your comment. We have updated the code and fixed errors about pre-train loading.
A simple colab file for medvit-large is attached.

[CIFAR100.zip](https://github.com/Omid-Nejati/MedViT/files/15480915/CIFAR100.zip)

**Comment by Betty-J** [comment]
> Thanks for your comment. We have updated the code and fixed errors about pre-train loading. A simple colab file for medvit-large is attached.
> 
> [CIFAR100.zip](https://github.com/Omid-Nejati/MedViT/files/15480915/CIFAR100.zip)
Hi, I think what AlKukharskii means is that in the code, the function defined for creating the model has a parameter called **_pretrained_** which is not actually used, as shown in the diagram.
![截屏2024-06-06 10 43 33](https://github.com/Omid-Nejati/MedViT/assets/23076443/dc3452db-2e86-4b50-b6d0-20d0a68edfee)

Additionally, I noticed that you have placed the loading of training weights in main.py, and specified that it should only occur when args.resume=True. This makes the overall structure seem a bit odd, or rather different from the usual steps of handling.
![截屏2024-06-06 10 46 01](https://github.com/Omid-Nejati/MedViT/assets/23076443/d2871292-acd9-4731-a813-222ccbe00733)

**Comment by kashiani** [comment]
Fixed in PR #14
